code/modules/mob/dead/observer/observer.dm

```diff
@@ -782,6 +782,35 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if(isobserver(user) && check_rights(R_SPAWN))
 		change_mob_type( /mob/living/carbon/human , null, null, TRUE) //always delmob, ghosts shouldn't be left lingering
 
+/mob/dead/observer/proc/ondemand_possess(mob/living/target)
+	var/possessable = target.ondemand_possessable
+	if(!isnum(possessable) || possessable <= 0)
+		return
+	var/original_key = target.key
+	var/control_message = control_mob_message(target)
+	var/confirmation = tgalert(src, control_message, "[target] can be possessed", "Yes", "No", Timeout=FALSE)
+	if(confirmation == "Yes" && !QDELETED(target))
+		if(target.key == original_key)
+			possessable = target.ondemand_possessable
+			if(isnum(possessable) && possessable > 0)
+				to_chat(target, "<span class='ghostalert'>Your mob has been taken over by a ghost!</span>")
+				message_admins("[key_name_admin(src)] has taken control of ([key_name_admin(target)])")
+				target.ghostize(0)
+				target.key = key
+				target.ondemand_possessable = 0
+			else
+				to_chat(src, "<span class='warning'>[target] is no longer possessable!</span>")
+		else
+			to_chat(src, "<span class='warning'>[target] was possessed by another ghost!</span>")
+
+/mob/dead/observer/examinate(atom/A as mob|obj|turf in view())
+	. = ..()
+	if(isliving(A))
+		var/mob/living/L = A
+		var/L_possessable = L.ondemand_possessable
+		if(isnum(L_possessable) && L_possessable > 0)
```

Comment:
just `if(L.ondemand_possessable)` is enough here, no need to declare an extra var